### PR TITLE
pythonPackages.pygam: init at 0.8.0

### PR DESCRIPTION
**Pull request by tetraptych**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @kalbasit

### CODE REVIEW

File: pkgs/development/python-modules/pygam/default.nix [hunk 1]
```diff
@@ -24,6 +24,6 @@ buildPythonPackage rec {
   meta = with lib; {
     description = "A Python implementation of Generalized Additive Models";
     homepage = https://github.com/dswah/pyGAM;
-    license = licenses.asl20; # unspecified
+    license = licenses.asl20; # Apache 2.0
```
Comment (FRidh) [inline pkgs/development/python-modules/pygam/default.nix]: this comment can be removed as it does not add any value

**Comment by FRidh** [comment]
@GrahamcOfBorg build python2.pkgs.pygam python3.pkgs.pygam

File: pkgs/development/python-modules/pygam/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+buildPythonPackage rec {
+  pname = "pygam";
+  version = "0.8.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0rli8w0fgfr6mxk5cx3n6i5zafi1j0abl2nslrrdxzmjm2p03bjw";
+  };
+
+  propagatedBuildInputs = [ future numpy pandas progressbar2 scipy ];
```
Comment (jonringer) [inline pkgs/development/python-modules/pygam/default.nix]: future package is only needed for python2
```suggestion
  propagatedBuildInputs = [ numpy pandas progressbar2 scipy ] ++ lib.optional isPy27 future;
```

File: pkgs/development/python-modules/pygam/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  checkInputs = [ pytest ];
+
+  meta = with lib; {
+    description = "A Python implementation of Generalized Additive Models";
+    homepage = https://github.com/dswah/pyGAM;
```
Comment (jonringer) [inline pkgs/development/python-modules/pygam/default.nix]: un-quoted url's are likely to be phased out, may as well change now
```suggestion
    homepage = "https://github.com/dswah/pyGAM";
```

**Comment by jonringer** [comment]
Please rebase the comments so that it's roughly 1 commit = 1 package in this case.

https://nixos.org/nixpkgs/manual/#submitting-changes-hotfixing-pull-requests

File: pkgs/development/python-modules/pygam/default.nix [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  pname = "pygam";
+  version = "0.8.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0rli8w0fgfr6mxk5cx3n6i5zafi1j0abl2nslrrdxzmjm2p03bjw";
+  };
+
+  propagatedBuildInputs = [ future numpy pandas progressbar2 scipy ];
+  checkInputs = [ pytest ];
```
Comment (risicle) [inline pkgs/development/python-modules/pygam/default.nix]: Note that the check phase doesn't actually succeed in finding/running any tests. To get these working you have to add a custom `checkPhase`:

```
  checkPhase = ''
    cd pygam/tests
    pytest
  '';
```

This passes successfully for me on non-nixos linux x86_64 & macos 10.13.

File: pkgs/development/python-modules/pygam/default.nix (hunk 2 above)
Comment (tetraptych) [reply pkgs/development/python-modules/pygam/default.nix]: Unfortunately it seems that the original library lists `future` as a requirement for all Python versions: https://github.com/dswah/pyGAM/blob/master/flit.ini#L16 .

After implementing the `isPy27` change, I receive the following error when running `nix-build . -A python3Packages.pygam`: 

```
Collecting future (from pygam==0.8.0)
  ERROR: Could not find a version that satisfies the requirement future (from pygam==0.8.0) (from versions: none)
ERROR: No matching distribution found for future (from pygam==0.8.0)
builder for '/nix/store/z6pcixliz8xlrbhwd76i074r0s36m59m-python3.7-pygam-0.8.0.drv' failed with exit code 1
error: build of '/nix/store/z6pcixliz8xlrbhwd76i074r0s36m59m-python3.7-pygam-0.8.0.drv' failed
```

Please advise.

File: pkgs/development/python-modules/pygam/default.nix (hunk 4 above)
Comment (tetraptych) [reply pkgs/development/python-modules/pygam/default.nix]: Done!

File: pkgs/development/python-modules/pygam/default.nix (hunk 1 above)
Comment (tetraptych) [reply pkgs/development/python-modules/pygam/default.nix]: Done.

File: pkgs/development/python-modules/pygam/default.nix (hunk 2 above)
Comment (jonringer) [reply pkgs/development/python-modules/pygam/default.nix]: oh, then add it back, and just leave a comment to the effect of `unfortunately future is listed in install_requires`

Comment (jonringer) [reply pkgs/development/python-modules/pygam/default.nix]: otherwise LGTM

**Comment by tetraptych** [comment]
Updated and rebased.

**Comment by kalbasit** [comment]
@GrahamcOfBorg build python2Packages.pygam python3Packages.pygam

**Comment by jonringer** [comment]
this seems to freeze when doing the python2 build

**Comment by jonringer** [comment]
I'm closing this PR due to lack of activity, if the PR author would like to pursue this matter again, please re-open the PR and address the concerns.

**Pull request closed without merging**